common/src/main/java/com/wynntils/utils/mc/ComponentUtils.java

```diff
@@ -168,4 +177,38 @@ public static MutableComponent makeObfuscated(
         }
         return newName;
     }
+
+    /**
+     * Adds a Wynntils chat banner to the left side of the provided text in the style of Wynncraft 2.1 chat banners.
+     * The formatting of the provided text is preserved.
+     *
+     * @param formattedText the formatted text to add the Wynntils chat banner to
+     * @return a {@code Component} holding the formatted text with the Wynntils chat banner added
+     */
+    public static Component addWynntilsBanner(Component component) {
+        Minecraft mc = Minecraft.getInstance();
+        List<FormattedText> lines = mc.font
+                .getSplitter()
+                .splitLines(
+                        StyledText.fromComponent(component).getString(PartStyle.StyleType.DEFAULT),
+                        ChatComponent.getWidth(mc.options.chatWidth().get())
+                                - mc.font.width(CHAT_BANNER_LINE_PREFIX + " "),
+                        Style.EMPTY);
+
+        MutableComponent output = Component.literal(CHAT_BANNER_FIRST_LINE)
+                .withStyle(CHAT_BANNER_STYLE)
+                .append("\n");
+
+        for (int i = 0; i < lines.size(); i++) {
+            output.append(Component.literal(CHAT_BANNER_LINE_PREFIX))
+                    .append(Component.literal(" ")
+                            .withStyle(Style.EMPTY.withFont(ResourceLocation.withDefaultNamespace("default")))
+                            .append(formattedTextToComponent(lines.get(i))));
+            if (i != lines.size() - 1) {
+                output.append("\n");
+            }
+        }
+
+        return output;
+    }
```

Comment:
Why not use `ComponentUtils#splitComponent` and add the formatting for every element in the list?

Reply:
It doesn’t work with wrapping the last line due to #2854.

Reply:
Okay, so we either fix that issue, or use StyledText. This solution does way too much work for essentially not much change for my liking, making me paranoid to accept this as a frequently used Util method.